rio_tiler/reader.py

```diff
@@ -474,7 +474,9 @@ def stats(
 
     hist_options = hist_options or {}
     return {
-        indexes[b]: raster_stats(data[b], percentiles=percentiles, **hist_options)
+        f"band{indexes[b]}": raster_stats(
```

Comment:
small but important breaking change! 

we now use only `str` as statistics key value, `{index} -> "band{index}"`

Reply:
Why is this important/necessary? Like I could see someone wanting to keep the band key as an integer so that they can then reference the original image again?

Reply:
it's better (IMO) for consistency for all the readers (e.g the multiBands reader use `band` B01 as key). 

also for pydantic model to support this we need to define the key type as Union[int, str] but this is risky because a string `"1"` will be casted to int ... having just one type (str) is mush safer. 

the idea is also to have a match with the band_description, band_metadata tuples `[("band1", "{band descr}" ]`.


When building frontend using output from `metadata` or `info`, having some consistency in the key type is also better.

Reply:
well In fact I'm going to re-change this AGAIN 🤦.
Setting for `band{}` each info (stats, band_description, band_metadata) doesn't help when you are working with simple COG (via COGReader) or Multiple Bands (via MultiBandsReader). 

For COG Reader we need to be able to set an `indexes=1,2,3,...` options while for MultiBands we need `bands=a,b,c`. The `key` should help to construct those.

Still we will enforce the key to be a string instead of an integer.